### PR TITLE
Compatability update

### PR DESCRIPTION
**Pull request by kearnsw**
Compatibility update to run with the latest version of the Rasa after the merging of Rasa Core and Rasa NLU.

### CODE REVIEW

File: requirements.txt
Comment (erohmensing) [inline requirements.txt]: let's pin these rasa and spacy versions. the latest rasa and whatever spacy it pins should do it! also, we definitely don't need all of these requirements (I think the creator of this repo just copied over from rasa), but there are some that we will need. Off the top of my head, these `typing`, `requests` and `rasa-sdk`. Happy to help you look for more.

Sidenote, we should also change the imports in `actions.py` to import from `rasa_sdk` instead of `rasa_core_sdk`.

File: Makefile
Comment (erohmensing) [inline Makefile]: I think after introducing the CLI, we can finally get rid of makefiles! Then I can also update the readme with those direct CLI commands instead of the make commands.

File: config.yml
Comment (erohmensing) [inline config.yml]:
```suggestion
  - name: MappingPolicy

```
So that we can use `action_restart` :)

**Review by erohmensing — APPROVED** [review]
Thank you for contributing your changes! There's still a few things to cover, I'm happy to help. Would you be able to give me push access to your fork so I can help finish up this PR directly?

File: requirements.txt
Comment (kearnsw) [reply requirements.txt]: I believe `typing` is part of the standard library. I've  checked all other imports in the repository and it seems  that `requests` and `rasa-sdk` are the only others requirements.

File: Makefile
Comment (kearnsw) [reply Makefile]: Feel free to remove these once the Readme is updated. I just wanted to ensure the code could be run using that instruction set.

Comment (erohmensing) [reply Makefile]: makes sense!

File: requirements.txt
Comment (erohmensing) [reply requirements.txt]: awesome, thank you for going through those!

Comment (erohmensing) [reply requirements.txt]: @kearnsw what is numpy required for?

File: Makefile
Comment (kearnsw) [reply Makefile]: I think one the areas my students were having the most trouble with was that the action server needed to still be running while using `rasa shell`, whereas the instructions in Readme could give the impression that they are run sequentially.

File: requirements.txt
Comment (kearnsw) [reply requirements.txt]: This can be deleted now as well I believe.

**Comment by kearnsw** [comment]
> Thank you for contributing your changes! There's still a few things to cover, I'm happy to help. Would you be able to give me push access to your fork so I can help finish up this PR directly?

You should now have access. Let me know if that is not the case. Thank you!

**Comment by liangjun-jiang** [comment]
Can someone merge this to master branch if everything looks good? I am not able to deploy the master branch to Rasa X due to all kinds of errors. 
You have a video tutorial series based on this repo but this repo doesn't work. It probably doesn't make that much sense.

**Pull request closed without merging**
